site/app/views/submission/HomeworkView.php

```diff
@@ -1210,14 +1209,198 @@ function downloadFile(file, path) {
+            }
+          });
+          event.preventDefault();
+        });
+        var regradeBox = document.getElementById("regradeBox");
+        var modal =document.getElementById("modal-container");
+        function showPopUp(){
+            regradeBox.style.display = "block";
+            modal.style.display = "block";
+        }  
+        function hidePopUp(){
+            regradeBox.style.display = "none";
+            modal.style.display = "none";
+        }; 
+      </script>
+HTML;
+      return $return;
+    }
+    public function showRegradeDiscussion($gradeable){
+         // if($this->core->getConfig()->isRegradeEnabled() == false){
+        //    return;
+        //  }
+          $thread_id = $this->core->getQueries()->getRegradeRequestID($gradeable->getId(), $gradeable->getUser()->getId());
+          $threads = $this->core->getQueries()->getRegradeDiscussion($thread_id);
+          $user = $this->core->getUser()->getId();
+          $first = true;
+          if($this->core->getUser()->accessGrading()){
+            $replyMessage = "Reply"; 
+            $replyPlaceHolder = "Enter your reply here";
+          }
+          else{
+            $replyMessage = "Request further TA/Instructor action"; 
+            $replyPlaceHolder = "If you believe you require more review, enter a reply here to request further TA/Instructor action...";
+          }
+          foreach ($threads as $thread) {
+            $class = ($this->core->getQueries()->isStaffPost($thread['user_id'])) ? "post_box important" : "post_box";
+            $id = $thread['id'];
+            $name = $this->core->getQueries()->getSubmittyUser($thread['user_id'])->getDisplayedFirstName();
+            $date = date_create($thread['timestamp']);
+            $content = $thread['content'];
+            if($first){
+              $class .= " first_post";
+              $first = false;                                      
+            }                                      
+            $function_date = 'date_format';                                      
+            $$return = <<<HTML
+            <div style="margin-top: 20px ">                                       
+              <div class = '$class' style="padding:20px;">                                       
+                <p>{$content}</p>                                      
+                <hr>                                       
+                <div style="float:right">                                      
+                  <b>{$name}</b> &nbsp                                       
+                {$function_date($date,"m/d/Y g:i A")}                                      
+                </div>                                       
+            </div>                                       
+HTML;
+          }
+        $return .= <<<HTML
+        <div style="padding:20px;">
+        <form method="POST" id="replyTextForm" action="{$this->core->buildUrl(array('component' => 'student',
```

Comment:
If there are no threads, $return is undefined at this point, throwing a warning.